tools/serverpod_cli/lib/src/runner/serverpod_command_runner.dart

```diff
@@ -6,41 +8,105 @@ import 'package:serverpod_cli/src/downloads/resource_manager.dart';
 import 'package:serverpod_cli/src/logger/logger.dart';
 import 'package:serverpod_cli/src/shared/environment.dart';
 import 'package:serverpod_cli/src/update_prompt/prompt_to_update.dart';
+import 'package:serverpod_cli/src/util/command_line_tools.dart';
 import 'package:serverpod_cli/src/util/exit_exception.dart';
 
 abstract class GlobalFlags {
-  static const developmentPrint = 'development-print';
+  static const quiet = 'quiet';
+  static const verbose = 'verbose';
+}
+
+typedef InitializeLoggerCallback = void Function(LogLevel);
+typedef PreCommandEnvironmentChecksCallback = Future<void> Function();
+
+Future<void> _preCommandEnvironmentChecks() async {
+  if (Platform.isWindows) {
+    log.warning(
+        'Windows is not officially supported yet. Things may or may not work '
+        'as expected.');
+  }
+
+  // Check that required tools are installed
+  if (!await CommandLineTools.existsCommand('dart')) {
+    log.error(
+        'Failed to run serverpod. You need to have dart installed and in your \$PATH');
+    throw ExitException();
+  }
+  if (!await CommandLineTools.existsCommand('flutter')) {
+    log.error(
+        'Failed to run serverpod. You need to have flutter installed and in your \$PATH');
+    throw ExitException();
+  }
+
+  if (!loadEnvironmentVars()) {
+    throw ExitException();
+  }
+
+  // Make sure all necessary downloads are installed
+  if (!resourceManager.isTemplatesInstalled) {
+    try {
+      await resourceManager.installTemplates();
+    } catch (e) {
+      log.error('Failed to download templates.');
+      throw ExitException();
+    }
+
+    if (!resourceManager.isTemplatesInstalled) {
+      log.error(
+          'Could not download the required resources for Serverpod. Make sure that you are connected to the internet and that you are using the latest version of Serverpod.');
+      throw ExitException();
+    }
+  }
 }
 
 class ServerpodCommandRunner extends CommandRunner {
   final Analytics _analytics;
   final bool _productionMode;
   final Version _cliVersion;
+  final InitializeLoggerCallback _loggerInitializationCallback;
+  final PreCommandEnvironmentChecksCallback
+      _preCommandEnvironmentChecksCallback;
 
   ServerpodCommandRunner(
```

Comment:
Suggestion: With this many parameters, it may be a better choice to use named parameters.